pystac/collection.py

```diff
@@ -678,6 +681,17 @@ def from_dict(
         if root:
             collection.set_root(root)
 
+        if (
+            CollectionVersionExtension.has_extension(collection)
+            and CollectionVersionExtension(collection).deprecated
+        ):
```

Comment:
Sorry to nitpick this, but I think the more "Pythonic" way would be something like:

```suggestion
        try:
            version = CollectionVersionExtension.ext(collection)
        except ExtensionNotImplemented:
             pass
        else:
            if version.deprecated:
                warnings.warn(
                    f"The collection '{collection.id}' is deprecated.",
                    DeprecatedWarning,
                )
```

Reply:
No need to apologize, your suggestion looks good. Implemented.